src/raftstore/store/snap.rs

```diff
@@ -995,11 +1002,15 @@ mod v2 {
                         .open(&cf_file.tmp_path));
                     cf_file.file = Some(f);
                 } else {
-                    // initialize sst file writer
-                    let handle = try!(snap.cf_handle(cf_file.cf));
-                    let env_opt = EnvOptions::new();
-                    let io_options = snap.get_db().get_options_cf(handle);
-                    let mut writer = SstFileWriter::new(env_opt, io_options);
+                    let handle = snap.cf_handle(cf_file.cf)?;
+                    let mut io_options = snap.get_db().get_options_cf(handle).clone();
+                    io_options.compression(self.compression_type);
+                    // in rocksdb 5.5.1, SstFileWriter will try to use bottommost_compression and
```

Comment:
Because literally bottommost_compression should only be the compression type of the bottommost layer. This part of logic in rocksdb is undocumented, we use compression and  reset bottommost_compression to get avoid possible logic changing of rocksdb in the future.